### PR TITLE
objects: Specify basic object types.

### PR DESCRIPTION
**Pull request by pepol**
Please take a look, ask if anything is not understandable and if you have suggestions for any other "built-in" type, let me know.

To: @andynet 
Cc: @JakubNvk 

---

This commit includes the first of the ImterraNet standards, the Object
Storage Specification, which will describe how exactly are the network's
objects stored.

This commit includes basic object types within the STD 1.

### CODE REVIEW

File: ObjectStorage [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+describe common object types in the next section.
+
+
+------------------------------
+SECTION 2 - BASIC OBJECT TYPES
+------------------------------
+
+This section describes format and usage of every common (built-in) type.  You
+may notice a similarity to git object types[2].  This is deliberate, since
+we found a git implemntation of Merkle DAG to be inspiring.
```
Comment (andynet) [inline ObjectStorage]: There is a typo in word implementation. Please fix it.

File: ObjectStorage [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Every objects can be defined as a data structure containing an object type
+string, a map of names to pointers to other objects (addressed by their hash),
+and a sequence of raw bytes up to 256 KiB serving data - the content of the
+object.
+
+However, this data structure should be considered _only_ as an abstract on
+which more concrete data structures are built.
+
+As we mentioned before, objects address other objects by their hash (which is
+actually a multihash[1]).  In order for this to form a Merkle DAG, it is
```
Comment (andynet) [inline ObjectStorage]: Please add some reference to Merkle DAG.

**Comment by andynet** [comment]
Please address the issues. Otherwise LGTM

File: ObjectStorage (hunk 1 above)
Comment (pepol) [inline ObjectStorage]: Fixed.

File: ObjectStorage (hunk 2 above)
Comment (pepol) [inline ObjectStorage]: Done.

**Comment by pepol** [comment]
@andynet all done, please take another look.
@JakubNvk please take a look as well and raise questions if you have any.

**Comment by andynet** [comment]
@pepol LGTM

**Pull request merged**